src/service/standard-actions-impl.js

```diff
@@ -103,6 +105,42 @@ export class StandardActions {
     );
   }
 
+  /**
+   * Handles initiliazing the theme mode.
+   *
+   * This methode needs to be called on page load to set the `amp-dark-mode`
+   * class on the body if the user prefers the dark mode.
+   */
+  initThemeMode_() {
+    if (this.prefersDarkMode_()) {
+      this.ampdoc.waitForBodyOpen().then((body) => {
+        const darkModeClass =
+          body.getAttribute('data-prefers-dark-mode-class') || 'amp-dark-mode';
+
+        body.classList.add(darkModeClass);
+      });
+    }
+  }
+
+  /**
+   * Checks whether the user prefers dark mode based on local storage and
+   * user's operating systen settings.
+   *
+   * @return {boolean}
+   */
+  prefersDarkMode_() {
+    try {
+      const themeMode = this.ampdoc.win.localStorage.getItem('amp-dark-mode');
+
+      if (themeMode) {
+        return 'yes' === themeMode;
+      }
+    } catch (e) {}
+
+    // LocalStorage may not be accessible
+    return this.ampdoc.win.matchMedia('(prefers-color-scheme: dark)').matches;
```

Comment:
Unfortunately our tests like to stub out `window` without creating a full API, so they're missing `matchMedia`. Doing this will fix:

```suggestion
    return this.ampdoc.win.matchMedia?.('(prefers-color-scheme: dark)').matches;
```

Reply:
Thanks, I have fixed the failing tests.